packages/design-system/src/components/Switch/Switch.tsx

```diff
@@ -1,94 +1,111 @@
 import * as React from 'react';
 
+import * as Switch from '@radix-ui/react-switch';
+import { composeEventHandlers } from '@strapi/ui-primitives';
 import { styled } from 'styled-components';
 
-import { Box } from '../Box';
+import { useControllableState } from '../../hooks/useControllableState';
 import { Flex } from '../Flex';
+import { Typography, TypographyComponent } from '../Typography';
 
-interface SwitchProps extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'onChange'> {
-  label: string;
-  onChange: React.MouseEventHandler<HTMLButtonElement>;
+interface SwitchProps extends Omit<Switch.SwitchProps, 'children'> {
   onLabel?: string;
   offLabel?: string;
-  selected?: boolean;
   visibleLabels?: boolean;
 }
 
-const Switch = React.forwardRef<HTMLButtonElement, SwitchProps>(
-  ({ label, onChange, onLabel = 'On', offLabel = 'Off', selected, visibleLabels = false, ...props }, ref) => {
+const SwitchImpl = React.forwardRef<HTMLButtonElement, SwitchProps>(
+  (
+    {
+      visibleLabels,
+      onLabel = 'On',
+      offLabel = 'Off',
+      onCheckedChange: onCheckedChangeProp,
+      checked: checkedProp,
+      defaultChecked,
```

Comment:
defaultChecked is a required prop?

Reply:
no, but `undefined` is fine as a default value :)